lib/pageflow/zencoder_video_output_definition.rb

```diff
@@ -14,98 +14,134 @@ def input_s3_url
     end
 
     def outputs
-      [
+      returned_outputs = [
         transferable(webm_high_definition),
         transferable(webm_medium_definition),
         transferable(mp4_high_definition),
         transferable(mp4_medium_definition),
         transferable(mp4_low_definition),
 
         hls_definitions,
-        non_transferable(smil_definition),
 
         thumbnails_definitions
-      ].flatten
+      ]
+      returned_outputs += non_transferable(smil_definition) unless skip_smil
+      if video_file.entry.feature_state('highdef_video_encoding')
+        returned_outputs += [transferable(mp4_4k_definition), transferable(mp4_fullhd_definition)]
+      end
+      returned_outputs.flatten
```

Comment:
Let's keep on using private methods here that might return `[]`. This is starting to read a bit worse than before